tools/database/src/memtrie.rs

```diff
@@ -78,3 +90,174 @@ impl LoadMemTrieCommand {
         Ok(())
     }
 }
+
+#[derive(clap::Parser)]
+pub struct SplitShardTrieCommand {
+    #[clap(long)]
+    shard_uid: ShardUId,
+    #[clap(long)]
+    boundary_account: AccountId,
+}
+
+impl SplitShardTrieCommand {
+    pub fn run(
+        &self,
+        home: &Path,
+        genesis_validation: GenesisValidationMode,
+    ) -> anyhow::Result<()> {
+        let env_filter = EnvFilterBuilder::from_env().verbose(Some("memtrie")).finish()?;
+        let _subscriber = default_subscriber(env_filter, &Default::default()).global();
+        let mut near_config = nearcore::config::load_config(&home, genesis_validation)
+            .unwrap_or_else(|e| panic!("Error loading config: {:#}", e));
+        near_config.config.store.load_memtries_for_tracked_shards = true;
+        let genesis_config = &near_config.genesis.config;
+        let chain_id = &genesis_config.chain_id;
+
+        // Create a database snapshot to avoid corrupting the existing database
+        println!("Creating database snapshot...");
+        let db_path =
+            near_config.config.store.path.as_ref().cloned().unwrap_or_else(|| home.join("data"));
+        let snapshot =
+            Snapshot::new(db_path.as_path(), &near_config.config.store, Temperature::Hot)?;
+        let snapshot_path =
+            snapshot.0.as_ref().ok_or_else(|| anyhow::anyhow!("Snapshot not created"))?;
```

Comment:
As we don't write anything to DB, we could skip it, especially it gives visible latency.
But maybe it is a good practice to always create snapshot beforehand. I don't know :)

Reply:
Actually, we do write something. `ReshardingManager` requires R+W database access. In particular, it saves the `ChunkExtra` objects for newly created shards.

Reply:
Wow. It's perfect then.